### PR TITLE
Fixed DOC-466 by renaming the feature to dictionary (5.7)

### PR DESCRIPTION
**Pull request by hrvojem**
Release notes updates to mention where the contribution for prefix index queries optimisation is coming from.
Mentioned that the gap locks detection is intended for storage engines like MyRocks.

### CODE REVIEW

File: doc/source/tokudb/tokudb_file_management.rst [hunk 1]
```diff
@@ -22,7 +22,7 @@ within a directory that represents a single database. This feature is enabled
 by default.
 
 In |Percona Server| :rn:`5.7.18-14` new :variable:`tokudb_dir_cmd` variable
-has been implemented that can be used to edit |TokuDB| files.
+has been implemented that can be used to edit the |TokuDB| dictionary.
```
Comment (george-lorch) [inline doc/source/tokudb/tokudb_file_management.rst]: Going to be a bit pedantic: In Tokutek nomenclature, a dictionary is an index or a single FT key/value FT file. The directory dictionary is what maps internal dictionary names to physical dictionary files on disk. See https://www.percona.com/blog/2016/09/27/tokudb-and-perconaft-database-file-management-part-1-of-2/ that explains the files. This new feature allows a user to manipulate the contents of the TokuDB/PerconaFT directory map.

So, if we want to be completely clear and accurate, we can reword this as "In |Percona Server| :rn:`5.7.18-14` new :variable:`tokudb_dir_cmd` variable has been implemented that can be used to edit the |TokuDB| directory mapping." and even perhaps link back to this blog post or some other doc content that has the same info.

File: doc/source/tokudb/tokudb_file_management.rst [hunk 2]
```diff
@@ -196,17 +196,17 @@ directory with a few steps:
 
 .. _editing_tokudb_files_with_tokudb_dir_cmd:
 
-Editing |TokuDB| files with :variable:`tokudb_dir_cmd` 
-------------------------------------------------------
+Editing |TokuDB| dictionary with :variable:`tokudb_dir_cmd`
```
Comment (george-lorch) [inline doc/source/tokudb/tokudb_file_management.rst]: Same dictionary/directory clarification.

File: doc/source/tokudb/tokudb_file_management.rst [hunk 3]
```diff
@@ -196,17 +196,17 @@ directory with a few steps:
...
 
 .. note:: 
 
   This feature is currently considered *Experimental*.
 
 In |Percona Server| :rn:`5.7.18-14` new :variable:`tokudb_dir_cmd` variable
-has been implemented that can be used to edit |TokuDB| files. **WARNING:** Use
-this variable only if you know what you're doing otherwise it **WILL** lead to
-data loss.
+has been implemented that can be used to edit the |TokuDB| dictionary.
```
Comment (george-lorch) [inline doc/source/tokudb/tokudb_file_management.rst]: Same...

File: doc/source/tokudb/tokudb_file_management.rst [hunk 4]
```diff
@@ -262,7 +262,7 @@ System Variables
      :dyn: Yes
      :vartype: String
 
-This variable is used to send commands to edit |TokuDB| directory files. 
+This variable is used to send commands to edit |TokuDB| dictionary.
```
Comment (george-lorch) [inline doc/source/tokudb/tokudb_file_management.rst]: Same...

**Review by george-lorch — CHANGES_REQUESTED** [review]

**Comment by hrvojem** [comment]
Changed term to  directory map as requested

**Review by george-lorch — APPROVED** [review]
